From edbc7986c6ec0428a82b1693827d2dec4424ed66 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 8 Dec 2010 04:09:22 +0000
Subject: [PATCH] block: Avoid ABI change from fix to blk_queue_physical_block_size()

We need to keep exporting a function with the old signature, so rename
the fixed version and add a macro and wrapper function to maintain
ABI and API compatibility.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 block/blk-settings.c   |   12 ++++++++++--
 include/linux/blkdev.h |    4 +++-
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/block/blk-settings.c b/block/blk-settings.c
index 7c7b8c1..f84d8ca 100644
--- a/block/blk-settings.c
+++ b/block/blk-settings.c
@@ -342,6 +342,14 @@ void blk_queue_logical_block_size(struct request_queue *q, unsigned short size)
 }
 EXPORT_SYMBOL(blk_queue_logical_block_size);
 
+#undef blk_queue_physical_block_size
+
+void blk_queue_physical_block_size(struct request_queue *q, unsigned short size)
+{
+	blk_queue_physical_block_size_fixed(q, size);
+}
+EXPORT_SYMBOL(blk_queue_physical_block_size);
+
 /**
  * blk_queue_physical_block_size - set physical block size for the queue
  * @q:  the request queue for the device
@@ -352,7 +360,7 @@ EXPORT_SYMBOL(blk_queue_logical_block_size);
  *   hardware can operate on without reverting to read-modify-write
  *   operations.
  */
-void blk_queue_physical_block_size(struct request_queue *q, unsigned int size)
+void blk_queue_physical_block_size_fixed(struct request_queue *q, unsigned int size)
 {
 	q->limits.physical_block_size = size;
 
@@ -362,7 +370,7 @@ void blk_queue_physical_block_size(struct request_queue *q, unsigned int size)
 	if (q->limits.io_min < q->limits.physical_block_size)
 		q->limits.io_min = q->limits.physical_block_size;
 }
-EXPORT_SYMBOL(blk_queue_physical_block_size);
+EXPORT_SYMBOL(blk_queue_physical_block_size_fixed);
 
 /**
  * blk_queue_alignment_offset - set physical block alignment offset
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 05f6018..1381980 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -932,7 +932,9 @@ extern void blk_queue_max_segment_size(struct request_queue *, unsigned int);
 extern void blk_queue_max_discard_sectors(struct request_queue *q,
 		unsigned int max_discard_sectors);
 extern void blk_queue_logical_block_size(struct request_queue *, unsigned short);
-extern void blk_queue_physical_block_size(struct request_queue *, unsigned int);
+extern void blk_queue_physical_block_size(struct request_queue *, unsigned short);
+extern void blk_queue_physical_block_size_fixed(struct request_queue *, unsigned int);
+#define blk_queue_physical_block_size blk_queue_physical_block_size_fixed
 extern void blk_queue_alignment_offset(struct request_queue *q,
 				       unsigned int alignment);
 extern void blk_limits_io_min(struct queue_limits *limits, unsigned int min);
-- 
1.7.2.3

